Libraries/Opc.Ua.Configuration/ApplicationInstance.cs

```diff
@@ -419,12 +421,26 @@ string productUri
         }
 
         /// <summary>
-        /// Delete the application certificate.
+        /// Checks for a valid application instance certificate.
         /// </summary>
-        public async Task DeleteApplicationInstanceCertificate(CancellationToken ct = default)
+        /// <param name="silent">if set to <c>true</c> no dialogs will be displayed.</param>
+        public Task<bool> CheckApplicationInstanceCertificates(
+            bool silent)
         {
+            return CheckApplicationInstanceCertificates(silent, CertificateFactory.DefaultLifeTime);
+        }
```

Comment:
if we do breaking changes, the lifetime and keysize should be ignored here.
Better were a setting in the config, for ECC its implicitly part of the cert type, but for RSA we may want to allow to set different sizes.
We need to think also about cert rotation for self signed certs.
Once a cert exists, it can be used even after it expires. Only an admin should roll it.